Extension/src/main.ts

```diff
@@ -99,11 +100,11 @@ export async function activate(context: vscode.ExtensionContext): Promise<CppToo
                     // If switching from disabled to enabled, we can continue activation.
                     currentIntelliSenseEngineValue = settings.intelliSenseEngine;
                     languageServiceDisabled = false;
-                    LanguageServer.activate();
+                    return LanguageServer.activate();
                 } else {
                     // We can't deactivate or change engines on the fly, so prompt for window reload.
                     reloadMessageShown = true;
-                    util.promptForReloadWindowDueToSettingsChange();
+                    return util.promptForReloadWindowDueToSettingsChange();
```

Comment:
Looking through the code to VS Code, it looks like it expects this activate to be an async.  (src\vs\workbench\api\common\extHostExtensionActivator.ts, in VS Code repo)  Wouldn't adding this return here, which awaits `showInformationMessage` under the hood, potentially cause us to never complete activation?

This is one reason why I'd prefer we do not ever propagate the return/await Promises that will only resolve when a user responds to UI.  Even for potential perf measurements - I don't think we will want to care how long it took the user to take an action, if ever.  It would be safer not to incur the risk of someone accidentally adding an await that may never resolve.

Reply:
It looks like I overlooked that this was happening in a lambda.  My concern about returning/awaiting promises from UI calls that may never be actions on, is still valid.  :)

Reply:
We do not ever need or want to give the caller the choice to wait for a UI operation to complete, in which we do not require any response from the user.  It's unnecessary to support this, and doing so introduces a way in which a simple mistake can lead to a deadlock.

Reply:
Ok, reverting